dothething/dothething/ClipsView.swift

```diff
@@ -12,20 +12,19 @@ struct ClipView: View {
     let clip: Clip
 
     var body: some View {
-        // thumbnail of video
-        Image(uiImage: clip.thumbnail)
-            .resizable()
-            .frame(width: 108, height: 192)
-            .cornerRadius(8)
-            .onTapGesture {
-                print("Tapped \(clip.url)")
-                Thinger.playVideo(videoUrl: clip.url)
+        GeometryReader { geometry in
+            if clip.thumbnail == UIImage() {
+                Image("Placeholder")
+                    .resizable()
+            } else {
+                Image(uiImage: clip.thumbnail)
+                    .resizable()
+                    .onTapGesture {
+                        print("Tapped \(clip.url)")
+                        Thinger.playVideo(videoUrl: clip.url)
+                    }
             }
```

Comment:
The new code introduced here checks if the clip's thumbnail is an empty UIImage, and if so, it displays a placeholder image instead. If the thumbnail is not empty, it displays the thumbnail and adds a tap gesture to it, which will open the video player.

This change improves the handling of situations where a video thumbnail may not be available, ensuring a consistent user experience by displaying a placeholder image. The code is easy to understand and follows good SwiftUI practices.